hastory-cli/src/Hastory/Cli/OptParse.hs

```diff
@@ -150,7 +150,7 @@ parseFlags =
        , help "URL of the central storage server"
        ]) <*>
   option
-    (Just . mkUsername . T.pack <$> str)
+    (Just . Username . T.pack <$> str)
```

Comment:
You want to use `maybeReader` here so that this can fail on invalid usernames.

Reply:
Done. Interesting behavior though. It seems that we are not discriminating between the case when the user provided an invalid username or when the user didn't provide a username at all.